base/shell/explorer/settings.cpp

```diff
@@ -20,85 +20,61 @@
 
 #include "precomp.h"
 
-TASKBAR_SETTINGS TaskBarSettings;
-const WCHAR szSettingsKey[] = L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer";
-const WCHAR szAdvancedSettingsKey[] = L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced";
+TaskbarSettings g_TaskbarSettings;
 
-VOID
-LoadTaskBarSettings(VOID)
+BOOL TaskbarSettings::Save()
 {
-    DWORD dwValue = NULL;
-    
-    LoadSettingDword(szAdvancedSettingsKey, L"TaskbarSizeMove", dwValue);
-    TaskBarSettings.bLock = (dwValue == 0);
-    
-    LoadSettingDword(szAdvancedSettingsKey, L"ShowSeconds", dwValue);
-    TaskBarSettings.bShowSeconds = (dwValue != 0);
-    
-    LoadSettingDword(szSettingsKey, L"EnableAutotray", dwValue);
-    TaskBarSettings.bHideInactiveIcons = (dwValue != 0); 
-    
-    LoadSettingDword(szAdvancedSettingsKey, L"TaskbarGlomming", dwValue);
-    TaskBarSettings.bGroupButtons = (dwValue != 0);
-    
-    TaskBarSettings.bShowQuickLaunch = TRUE;    //FIXME: Where is this stored, and how?
-    
-    /* FIXME: The following settings are stored in stuckrects2, do they have to be load here too? */
-    TaskBarSettings.bShowClock = TRUE;
-    TaskBarSettings.bAutoHide = FALSE;
-    TaskBarSettings.bAlwaysOnTop = FALSE;
+    SHSetValueW(hkExplorer, NULL, L"EnableAutotray", REG_DWORD, &bHideInactiveIcons, sizeof(bHideInactiveIcons));
+    SHSetValueW(hkExplorer, L"Advanced", L"ShowSeconds", REG_DWORD, &bShowSeconds, sizeof(bShowSeconds));
+    SHSetValueW(hkExplorer, L"Advanced", L"TaskbarGlomming", REG_DWORD, &bGroupButtons, sizeof(bGroupButtons));
+    BOOL bAllowSizeMove = !bLock;
+    SHSetValueW(hkExplorer, L"Advanced", L"TaskbarSizeMove", REG_DWORD, &bAllowSizeMove, sizeof(bAllowSizeMove));
+    sr.cbSize = sizeof(sr);
+    SHSetValueW(hkExplorer, L"StuckRects2", L"Settings", REG_BINARY, &sr, sizeof(sr));
 
+    /* AutoHide writes something to HKEY_CURRENT_USER\Software\Microsoft\Internet Explorer\Desktop\Components\0 figure out what and why */
```

Comment:
Nitpick: Wouldn't it be better to have "TODO" in the comment to it's easier to find in the future?

Reply:
Noted

Reply:
Iirc there was a TODO when I added the comment. :p